clientconn_test.go

```diff
@@ -701,3 +702,43 @@ func TestGetClientConnTarget(t *testing.T) {
 		t.Fatalf("Target() = %s, want %s", cc.Target(), addr)
 	}
 }
+
+type backoffForever struct{}
+
+func (b backoffForever) Backoff(int) time.Duration { return time.Duration(math.MaxInt64) }
+
+func TestResetConnectBackoff(t *testing.T) {
+	defer leakcheck.Check(t)
+	dials := make(chan struct{})
+	dialer := func(string, time.Duration) (net.Conn, error) {
+		dials <- struct{}{}
+		return nil, errors.New("failed to fake dial")
+	}
+	cc, err := Dial("any", WithInsecure(), WithDialer(dialer), withBackoff(backoffForever{}))
+	if err != nil {
+		t.Fatalf("Dial() = _, %v; want _, nil", err)
+	}
+	defer cc.Close()
+	select {
+	case <-dials:
+		break
```

Comment:
Nit: this break is unnecessary, right?

Reply:
Technically, yes it's not necessary.  But it looks weird to a C programmer without it.  Removed anyway, this is Go.